docs/notes.md

```diff
@@ -603,4 +603,242 @@ sends a pair \\((L\_j,R\_j)\\) of points at each step \\(j = k\dots1\\). An
+         &+   {\langle {\mathbf{a}}\_{Lj}, {\mathbf{a}}\_{Rj} \circ {\mathbf{y}}^{n} \cdot y_j \rangle} 
+\end{aligned}
+\\]
+
+Combining inner products
+------------------------
+
+We combine the terms in the last statement into a single inner product, using the same technique as in the single-value range proof. We will not reproduce the math here, only the end result:
+
+\\[
+\begin{aligned}
+ \delta_j(y,z) &= (z - z^{2}) \cdot {\langle {\mathbf{1}}, {\mathbf{y}}^{n} \cdot y_j \rangle} - z^{3} z_j \cdot {\langle {\mathbf{1}}, {\mathbf{2}}^{n} \rangle}\\\\
+ z^{2}z_j \cdot v + \delta_j(y,z) &= {\langle {\mathbf{a}}\_{Lj} - z {\mathbf{1}}, {\mathbf{y}}^{n} \cdot y_j \circ ({\mathbf{a}}\_{Rj} + z {\mathbf{1}}) + z^{2} z_j \cdot {\mathbf{2}}^{n} \rangle}
+\end{aligned} 
+\\]
+
+Blinding the inner product
+--------------------------
+
+The prover chooses vectors of blinding factors
+\\[
+{\mathbf{s}}\_{Lj}, {\mathbf{s}}\_{Rj} \\;{\xleftarrow{\\$}}\\; {\mathbb Z\_p}^{n}
+\\]
+and uses them to construct blinded vector polynomials:
+\\[
+\begin{aligned}
+  {\mathbf{l}}\_j(x) &= ({\mathbf{a}}\_{Lj} + {\mathbf{s}}\_{Lj} x) - z {\mathbf{1}} & \in {\mathbb Z\_p}[x]^{n}  \\\\
+  {\mathbf{r}}\_j(x) &= {\mathbf{y}}^{n} \cdot y_j \circ \left( ({\mathbf{a}}\_{Rj} + {\mathbf{s}}\_{Rj} x\right)  + z {\mathbf{1}}) + z^{2} z_j {\mathbf{2}}^{n} &\in {\mathbb Z\_p}[x]^{n} 
+\end{aligned}
+\\]
+The vector polynomials \\({\mathbf{l}}\_j(x)\\) and \\({\mathbf{r}}\_j(x)\\) are related to \\(t(x)\\) by the following equations:
+\\[
+\begin{aligned}
+  t_j(x) &= {\langle {\mathbf{l}}\_j(x), {\mathbf{r}}\_j(x) \rangle} \\\\
+  &= t\_{j0} + t\_{j1} x + t\_{j2} x^{2} \\\\
+  t\_{j0} &= {\langle {\mathbf{l}}\_{j0}, {\mathbf{r}}\_{j0} \rangle} \\\\
+  &= {\langle {\mathbf{a}}\_{Lj} - z {\mathbf{1}}, {\mathbf{y}}^{n} \cdot y_j \circ ({\mathbf{a}}\_{Rj} + z {\mathbf{1}}) + z^{2} z_j \cdot {\mathbf{2}}^{n} \rangle} \\\\
+  t\_{j1} &= {\langle {\mathbf{l}}\_{j1}, {\mathbf{r}}\_{j0} \rangle} + {\langle {\mathbf{l}}\_{j0}, {\mathbf{r}}\_{j1} \rangle} \\\\
+  t\_{j2} &= {\langle {\mathbf{l}}\_{j1}, {\mathbf{r}}\_{j1} \rangle} \\\\
+\end{aligned}
+\\]
+
+Proving that \\(t(x)\\) is correct
+----------------------------------
+
+Proving that \\(t\_j(x)\\) is correct means proving that
+\\({\mathbf{l}}\_j(x)\\), \\({\mathbf{r}}\_j(x)\\) are correctly formed, and that
+\\(t_j(x) = {\langle {\mathbf{l}}\_j(x), {\mathbf{r}}\_j(x) \rangle}\\).
+
+We can combine the statements about \\(t_j(x)\\), \\({\mathbf{l}}\_j(x)\\), and \\({\mathbf{r}}\_j(x)\\) from all `m` parties in the following manner:
+
+\\[
+\begin{aligned}
+  t(x) &= \sum_{j=0}^{m-1} t_j(x)\\\\
+  {\mathbf{l}}(x) &= {\mathbf{l}}\_{j=0}(x) || {\mathbf{l}}\_{j=1}(x) || \dots || {\mathbf{l}}\_{j=m-1}(x) \\\\
+  {\mathbf{r}}(x) &= {\mathbf{r}}\_{j=0}(x) || {\mathbf{r}}\_{j=1}(x) || \dots || {\mathbf{r}}\_{j=m-1}(x) \\\\
```

Comment:
This should maybe be
``` 
 {\mathbf{r}}(x) &= {\mathbf{r}}\_{0}(x) || {\mathbf{r}}\_{1}(x) || \dots || {\mathbf{r}}\_{m-1}(x) \\\\ 
```
instead of 
```
{\mathbf{r}}(x) &= {\mathbf{r}}\_{j=0}(x) || {\mathbf{r}}\_{j=1}(x) || \dots || {\mathbf{r}}\_{j=m-1}(x) \\\\ 
```

Reply:
hm, I was trying to find some way to differentiate `r_{j=0}` from the `r_0` and `r_1` values that are the exponents of the `r` polynomial (used in lines 692-697)... I think that this is clearer that it is the `j` value that is `0`, but I think the subscript overloading is inherently clunky :/

Reply:
Ah, I see what you mean... you're right, this is awkward. Maybe we can think of a better/tweaked notation?

Reply:
what about `r_(j)` for `j`-party subscripts in general, and in this case it would be `r_(0)`, `r_(1)` which is distinct from `r_0`, `r_1`?

Reply:
That seems reasonable! It is still a bit awkward, but I'm at a loss to think of something that's better.